pkg/api/validation/validation.go

```diff
@@ -1442,44 +1442,59 @@ func ValidateLimitRange(limitRange *api.LimitRange) errs.ValidationErrorList {
 		limitTypeSet[limit.Type] = true
 
 		keys := util.StringSet{}
-		min := map[string]int64{}
-		max := map[string]int64{}
-		defaults := map[string]int64{}
-		defaultRequests := map[string]int64{}
+		min := map[string]resource.Quantity{}
+		max := map[string]resource.Quantity{}
+		defaults := map[string]resource.Quantity{}
+		defaultRequests := map[string]resource.Quantity{}
 
 		for k := range limit.Max {
 			allErrs = append(allErrs, validateResourceName(string(k), fmt.Sprintf("spec.limits[%d].max[%s]", i, k))...)
 			keys.Insert(string(k))
 			q := limit.Max[k]
-			max[string(k)] = q.Value()
+			max[string(k)] = q
 		}
 		for k := range limit.Min {
 			allErrs = append(allErrs, validateResourceName(string(k), fmt.Sprintf("spec.limits[%d].min[%s]", i, k))...)
 			keys.Insert(string(k))
 			q := limit.Min[k]
-			min[string(k)] = q.Value()
+			min[string(k)] = q
 		}
 		for k := range limit.Default {
 			allErrs = append(allErrs, validateResourceName(string(k), fmt.Sprintf("spec.limits[%d].default[%s]", i, k))...)
 			keys.Insert(string(k))
 			q := limit.Default[k]
-			defaults[string(k)] = q.Value()
+			defaults[string(k)] = q
 		}
 		for k := range limit.DefaultRequest {
 			allErrs = append(allErrs, validateResourceName(string(k), fmt.Sprintf("spec.limits[%d].defaultRequest[%s]", i, k))...)
 			keys.Insert(string(k))
 			q := limit.DefaultRequest[k]
-			defaultRequests[string(k)] = q.Value()
+			defaultRequests[string(k)] = q
 		}
 		for k := range limit.MaxLimitRequestRatio {
 			allErrs = append(allErrs, validateResourceName(string(k), fmt.Sprintf("spec.limits[%d].maxLimitRequestRatio[%s]", i, k))...)
 		}
 
 		for k := range keys {
-			minValue, minValueFound := min[k]
-			maxValue, maxValueFound := max[k]
-			defaultValue, defaultValueFound := defaults[k]
-			defaultRequestValue, defaultRequestValueFound := defaultRequests[k]
+			minQuantity, minValueFound := min[k]
+			maxQuantity, maxValueFound := max[k]
+			defaultQuantity, defaultValueFound := defaults[k]
+			defaultRequestQuantity, defaultRequestValueFound := defaultRequests[k]
+
+			// we need to normalize to the most common form of precision
+			minValue := minQuantity.Value()
+			maxValue := maxQuantity.Value()
+			defaultValue := defaultQuantity.Value()
+			defaultRequestValue := defaultRequestQuantity.Value()
+			if minValue <= resource.MaxMilliValue &&
+				maxValue <= resource.MaxMilliValue &&
+				defaultValue <= resource.MaxMilliValue &&
+				defaultRequestValue <= resource.MaxMilliValue {
```

Comment:
Yeah, I recommend adding a .Cmp method and using it instead of using Value/MilliValue.